### PR TITLE
feat: applied DIY coverage tool on strassen() and added coverage test…

### PR DESCRIPTION
**Pull request by AmandaStromdahl**
…s #38

### CODE REVIEW

File: coverage_analysis/functions_analysed/strassen_coverage.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        [a[i][j] for j in range(mid, matrix_length)] for i in range(mid, matrix_length)
+    ]
+
+    top_left = [[a[i][j] for j in range(mid)] for i in range(mid)]
+    bot_left = [[a[i][j] for j in range(mid)] for i in range(mid, matrix_length)]
+
+    return top_left, top_right, bot_left, bot_right
+
+def strassen(matrix1: list, matrix2: list, coverage: CoverageData) -> list:
+    if matrix_dimensions(matrix1)[1] != matrix_dimensions(matrix2)[0]:
```
Comment (bastienfaivre) [inline coverage_analysis/functions_analysed/strassen_coverage.py:107]: check the requirement 3 under 3.5.1, each time you've a "if" statement, if the "else" is missing, you need to add it. i.e you can just add:
`else:
    coverage.log_branch(type, x)
`

**Review by bastienfaivre — COMMENTED** [review]
Please check my comment and let me know if I'm wrong with my understanding of the point 3 under 3.5.1 in the assignment

File: coverage_analysis/functions_analysed/strassen_coverage.py (hunk 1 above)
Comment (AmandaStromdahl) [reply coverage_analysis/functions_analysed/strassen_coverage.py:107]: Sure, I'll add that.

**Comment by AmandaStromdahl** [comment]
I updated strassen() by adding missing else-statements.

**Comment by bastienfaivre** [comment]
LGTM

**Pull request merged**